### ISSUE TITLE
Stop prefixing `Entity` with `Arg_`

### ISSUE THREAD

**Issue by JDGrimes** [WordPoints/hooks-api#20]
## Origin

The Entity portion of the Hooks API started out being called the "Args API". The name was derived from the fact that the values for the Entities come from the arguments that an action was called with. At that time, Entities were not yet differentiated from Attributes and Relationships, and these were all called "Args" collectively. However, now the Entities have had their position as first-class citizens sealed, while Attributes and Relationships are only Entity children.
## Current State

There is still some overlap between the interface for these three elements, and so there is still a base `WordPoints_Hook_Arg` class and a `WordPoints_Hook_ArgI` interface. And perhaps we'll even want to keep those.
## Reason for Change

The reason for renaming the API, is because it has many uses beyond just encapsulating action args. It isn't a sub-part of the Actions API, or we would call them "Action Args" instead. It is a first-class part of the Hooks API, and deserves to be recognized as such. Yes, it is used by the Action API, but it is also used by just about every other part of the API, and its usage doesn't have to be directly related to actions at all. In fact, the Actions are a very low-level API which is practically hidden from all other parts, while Entities are a much more prominent piece of the API—I'd go so far as to say that they _make_ the API. It certainly wouldn't be possible for any of the other pieces to work at all without them.

Which is why all of the `Args_` everywhere needs to go. The exception would be where it is specifically relating to Action args, and only relating to Action args.

**Comment by JDGrimes** [comment]
Really, the Entity API surely has uses beyond the Hooks API entirely. But I'm not sure I feel comfortable with dropping the `Hook_` prefix too. It just kind of seems like we should wait until the Entity API is more mature, though really I'm not sure why.

In truth, the entire API could be moved out of the `hooks` sub-heading, though that seems even scarier.

**Comment by JDGrimes** [comment]
I believe we do need to keep the common interface between the Entity and Entity children (Atts and Relationships). The next order of business then is to decide if we can rename the `WordPoints_Hook_ArgI` interface to something different. The problem is encompassing all of these three things (and possibly others), which is why such a generic term was used in the first place.

We could change the name to  `WordPoints_Hook_EntityI`, but then it seems strange that Entity children. Perhaps it is time to bring out the old "ish". We could name it `WordPoints_Hook_EntityishI`. It's kind of strange, but it does the job.

Basically I'm looking for something that will clearly be related to the Entity API, but which also shows that the interface isn't used only for the Entities themselves, but also for their children.

**Comment by JDGrimes** [comment]
I'm actually starting to feel that the `Hook_` prefix is out of place in this API. Having a Hook Reactor, Hook Reaction, Hook Event, Hook Action, or Hook Extension makes sense. A Hook Entity? Well, that makes it sound like the Entity is a Hook—which it isn't (it could be actually, but that's beside the point). Like I said above the Entity API is not really bound to the Hooks API in some way. The two are very decoupled, so much so that I see no reason why the Entity API could be easily used without any other part of the Hooks API. Which is what leads me to think that maybe we really should seriously think about taking the leap and making them sisters instead of mother and daughter.

**Comment by JDGrimes** [comment]
I've dropped the `Hook_Arg_` prefix from the Entity related stuff. It is much better now.
